src/pyrobot/core.py

```diff
@@ -48,83 +41,140 @@
 import os
 
 
-
 import libtmux
 import os
 import os.path as osp
 
+
 def make_module(cfg):
-    module =  instantiate(cfg.object, configs=cfg.conf)
+    module = instantiate(cfg.object, configs=cfg.conf)
     return module
 
 
-def make_robot(robot_cfg, ns='', overrides=[], ros_launch_manager=None):
-
-    # TODO: add check if the robot name passed is correct        
-    
+def make_robot(robot_cfg, ns="", overrides=[], ros_launch_manager=None):
```

Comment:
Could we duck-type the robot dictionary object to behave like a class? i.e can we wrap the dictionary in an object such that a call like robot['arm]['gripper'] =  = robot.arm.gripper ?